main/src/ca/uwaterloo/flix/language/phase/Redundancy.scala

```diff
@@ -795,6 +797,12 @@ object Redundancy {
   private def isUselessExpression(exp: Expression): Boolean =
     exp.pur == Type.Pure
 
+  /**
+    * Returns true if the expression is not pure and not unit type.
+    */
+  private def isImpureDiscardedvalue(exp: Expression): Boolean =
```

Comment:
isImpureDiscardedvalue -> isImpureDiscardedValue

Reply:
I think we should allow the expression to be pure. What do you think? Or was I voted down? :)

Reply:
> I think we should allow the expression to be pure. What do you think? Or was I voted down? :)

I voted against it, but I don't know what the consensus was.